### PR TITLE
docs: implement module preloading strategy

### PR DESCRIPTION
**Pull request by rubenperegrina**

### CODE REVIEW

**Comment by santoshyadavdev** [comment]
Hey @rubenperegrina thank you for contribution, can you also add the way we can do it using StandAlone API, this is a great contribution

**Comment by rubenperegrina** [comment]
Hey @santoshyadavdev , I just uploaded it using the StandAlone API, check it out

File: src/pages/snippets/preloading-strategy.mdx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  },
+  {
+    path: "about",
+    loadComponent: () =>
+      import("./about/about.component").then((m) => m.AboutComponent),
+  },
+];
+
+//In your main.ts add:
+bootstrapApplication(AppComponent, {
```
Comment (santoshyadavdev) [inline src/pages/snippets/preloading-strategy.mdx:68]: this a solution for backward compatibility, with new API you can do 

` providers: [
    provideRouter(
      routes
    ),`

File: src/pages/snippets/preloading-strategy.mdx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  ],
+  exports: [RouterModule],
+})
+export class AppRoutingModule {}
+
+//StandAlone API version
+//In your main.ts add:
+bootstrapApplication(AppComponent, {
+  providers: [
+    importProvidersFrom(RouterModule.forRoot(routes, {
```
Comment (santoshyadavdev) [inline src/pages/snippets/preloading-strategy.mdx:95]: with new API you can do

` providers: [
    provideRouter(
      routes,
      withPreloading(PreloadAllModules)
    ),`

File: src/pages/snippets/preloading-strategy.mdx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+export class PreloadingStrategyService implements PreloadingStrategy {
+  private preloadedModules: string[] = [];
+
+  preload(route: Route, load: () => Observable<any>): Observable<unknown> {
+    if (route.data && route.data["preload"] && route.path) {
+      this.preloadedModules.push(route.path);
+      return load();
+    } else {
+      return of(null);
+    }
```
Comment (santoshyadavdev) [inline src/pages/snippets/preloading-strategy.mdx:116]: @Injectable({
  providedIn:'root'
})
export class PreloadingStrategyService implements PreloadingStrategy {
  private preloadedModules: string[] = [];
  preload(route: Route, load: () => Observable<any>): Observable<unknown> {
    if (route.data && route.data['preload'] && route.path) {
      this.preloadedModules.push(route.path);
      return load();
    } else {
      return of(null);
    }
  }
}

File: src/pages/snippets/preloading-strategy.mdx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    }),
+  ],
+  exports: [RouterModule],
+})
+export class AppRoutingModule {}
+
+//StandAlone API version
+//In your main.ts add:
+bootstrapApplication(AppComponent, {
+  providers: [
```
Comment (santoshyadavdev) [inline src/pages/snippets/preloading-strategy.mdx:147]: providers: [
    provideRouter(
      routes,
      withPreloading(PreloadingStrategyService),
    )

**Review by santoshyadavdev — COMMENTED** [review]
Thank you for making changes, lets's use the new standalone API.

**Review by santoshyadavdev — APPROVED** [review]

**Pull request merged**